internal/handlers/device.go

```diff
@@ -43,14 +44,31 @@ func (api *API) ListDevices(c *gin.Context) {
 	ctx, span := tracer.Start(c.Request.Context(), "ListDevices")
 	defer span.End()
 	devices := make([]models.Device, 0)
-	result := api.db.WithContext(ctx).Scopes(FilterAndPaginate(&models.Device{}, c)).Find(&devices)
+
+	result := api.db.WithContext(ctx).Scopes(
+		api.DeviceIsVisibleToCurrentUser(c),
+		FilterAndPaginate(&models.Device{}, c),
+	).Find(&devices)
+
 	if result.Error != nil {
 		c.JSON(http.StatusInternalServerError, gin.H{"message": "error fetching keys from db"})
 		return
 	}
 	c.JSON(http.StatusOK, devices)
 }
 
+func (api *API) DeviceIsVisibleToCurrentUser(c *gin.Context) func(db *gorm.DB) *gorm.DB {
+	return func(db *gorm.DB) *gorm.DB {
+		userId := c.Value(gin.AuthUserKey).(string)
+		// this could potentially be driven by rego output
+		if api.dialect == database.DialectSqlLite {
+			return db.Where("user_id = ? OR organization_id in (SELECT organization_id FROM user_organizations where user_id=?)", userId, userId)
+		} else {
+			return db.Where("user_id = ? OR organization_id::text in (SELECT organization_id::text FROM user_organizations where user_id=?)", userId, userId)
```

Comment:
I assume a user should be able to see all the devices he owns and all the devices owned by orgs that he is member of.  If this should be jus the ones he owns.. then this list query gets much simpler.